source/intercept/source/collect/Intercept.cc

```diff
@@ -110,11 +144,14 @@ namespace ic {
     { }
 
     rust::Result<ps::CommandPtr> Intercept::command(const flags::Arguments &args, const char **envp) const {
-        const auto execution = capture_execution(args, sys::env::from(envp));
-        const auto session = Session::from(args, envp);
-        const auto reporter = Reporter::from(args);
-
-        return rust::merge(execution, session, reporter)
+        return into_configuration(args)
+                .and_then<std::tuple<Execution, Session::Ptr, Reporter::Ptr>>([&envp](const auto& configuration) {
```

Comment:
Passing this long template argument is a bit ugly, but that is the only way to make it work.
Maybe with deduction guides it's possible to make the compiler deduct the type, but deduction guides are an obscure part of C++ and I don't know them very well